File: Practicals/prac_4_1.py
import os
import time

def main():
    pid = os.fork()  # Create a new process

    if pid == 0:
        # Child Process
        print(f"Child Process, PID: {os.getpid()}")  # Child prints its PID
        time.sleep(2)  # Simulate work with sleep
        print("Child Task Done")
    else:
        # Parent Process
        print(f"Parent Process, PID: {os.getpid()}")  # Parent prints its PID
        os.wait()  # Wait for the child process to finish
        print("Parent Task Done")

if __name__ == "__main__":
    main()

File: Practicals/prac_4_1.cpp
#include <iostream>
#include <unistd.h>

int main() {
    pid_t pid = fork(); // Create a new process

    if (pid < 0) {
        std::cerr << "Fork failed\n";
        return 1;
    }

    // Both parent and child execute the same code after fork()
    std::cout << "Process ID: " << getpid() << ", Parent ID: " << getppid() << "\n";

    // Simulating work common to both processes
    sleep(1); // Both parent and child will sleep for 1 second
    std::cout << "Task Done by Process ID: " << getpid() << "\n";

    return 0;
}
